j5/boards/base.py

```diff
@@ -74,15 +74,16 @@ class BoardGroup:
     def __init__(self, board: Board, backend: Backend):
         self.board_class: Board = board
         self._backend: Backend = backend
-        self._iterator_counter: int = 0
         self.boards: Dict[str, Board] = {}
 
         self.update_boards()
 
     def update_boards(self) -> None:
         """Update the boards in this group to see if new boards have been added."""
         self.boards: Dict[str, Board] = {}
-        for board in self.board_class.discover(self._backend):
+        discovered_boards = self.board_class.discover(self._backend)
+        discovered_boards.sort(key=lambda board: board.serial)
+        for board in discovered_boards:
```

Comment:
What is the minimum version of Python we are supporting?

Dictionarys being ordered was introduced to the Python standard in 3.7 and as a side effect to the CPython implementation in 3.6. If we wanted to support Python 3.5 (which is the first to allow syntactical type hints) or non CPython 3.6 we'd have to change `self.boards` to be an `OrderdDict` from `collections`

Reply:
Good point, we should really specify this somewhere. I've created #79.